pymongo/encryption.py

```diff
@@ -229,7 +229,10 @@ def insert_data_key(self, data_key):
         """
         raw_doc = RawBSONDocument(data_key, _KEY_VAULT_OPTS)
         data_key_id = raw_doc.get("_id")
-        if not isinstance(data_key_id, uuid.UUID):
+        if isinstance(data_key_id, Binary):
+            if data_key_id.subtype != UUID_SUBTYPE:
+                raise TypeError("data_key _id must have a UUID subtype")
+        elif not isinstance(data_key_id, uuid.UUID):
```

Comment:
We should _only_ accept Binary, never UUID here. Let's also update the return below:
```
- return Binary(data_key_id.bytes, subtype=UUID_SUBTYPE) 
+ return data_key_id
```